components/MentorshipsList/index.js

```diff
@@ -1,12 +1,92 @@
 import { Alert, AlertIcon } from "@chakra-ui/alert";
 import { SimpleGrid, Text } from "@chakra-ui/layout";
+import { useDisclosure } from "@chakra-ui/hooks";
+import { useState, useRef } from "react";
+import { useToast } from "@chakra-ui/toast";
 import useSWR from "swr";
 import MentorshipCard from "./MentorshipCard";
+import TupuAlertDialog from "../TupuAlertDialog";
 
 const fetcher = (...args) => fetch(...args).then((res) => res.json());
 
 const MentorshipsList = () => {
-  const { data, error } = useSWR("/api/mentorships", fetcher);
+  const { data, error, mutate } = useSWR("/api/mentorships", fetcher);
+
+  const [actionDialogParams, setActionDialogParams] = useState(null);
+  const [isActionDialogLoading, setActionDialogLoading] = useState(false);
+  const { isOpen: isAlertDialogOpen, onOpen: onAlertDialogOpen, onClose: onAlertDialogClose } = useDisclosure();
+  const cancelRef = useRef();
+  const toast = useToast();
+
```

Comment:
Should we move the `mutateRelationshipsList` and `changeMentorshipRelationship` functions into a separate file? At the moment that will be declared again each time the component renders

Reply:
I created a follow up ticket: https://github.com/tupuio/tupuapp/issues/118